extensions/admin-tool-ext-win/src/test/index.ts

```diff
@@ -3,28 +3,60 @@
  *  Licensed under the Source EULA. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-const path = require('path');
-const testRunner = require('vscode/lib/testrunner');
+import * as Mocha from 'mocha';
+import * as path from 'path';
+import * as glob from 'glob';
 
 const suite = 'Database Admin Tool Extensions for Windows';
 
-const options: any = {
-	ui: 'bdd',
-	useColors: true,
-	timeout: 600000
-};
-
-if (process.env.BUILD_ARTIFACTSTAGINGDIRECTORY) {
-	options.reporter = 'mocha-multi-reporters';
-	options.reporterOptions = {
-		reporterEnabled: 'spec, mocha-junit-reporter',
-		mochaJunitReporterReporterOptions: {
-			testsuitesTitle: `${suite} ${process.platform}`,
-			mochaFile: path.join(process.env.BUILD_ARTIFACTSTAGINGDIRECTORY, `test-results/${process.platform}-${suite.toLowerCase().replace(/[^\w]/g, '-')}-results.xml`)
-		}
+export function run(): Promise<void> {
+	const options: Mocha.MochaOptions = {
+		ui: 'bdd',
+		timeout: 600000,
+		grep: process.env.ADS_TEST_GREP
 	};
-}
 
-testRunner.configure(options);
+	if (process.env.BUILD_ARTIFACTSTAGINGDIRECTORY) {
+		options.reporter = 'mocha-multi-reporters';
+		options.reporterOptions = {
+			reporterEnabled: 'spec, mocha-junit-reporter',
+			mochaJunitReporterReporterOptions: {
+				testsuitesTitle: `${suite} ${process.platform}`,
+				mochaFile: path.join(process.env.BUILD_ARTIFACTSTAGINGDIRECTORY, `test-results/${process.platform}-${suite.toLowerCase().replace(/[^\w]/g, '-')}-results.xml`)
+			}
+		};
+	}
+
+	const mocha = new Mocha(options);
+	mocha.useColors(true);
+
+	if (process.env.ADS_TEST_INVERT_GREP) {
```

Comment:
if environment variable ADS_TEST_INVERT_GREP is set to 0, the tests will start be inverted. Do you want that behavior?

Reply:
Yes, the intention of this was that if it's set at all then the grep would be inverted. I don't really see value in trying to parse out the option value.